src/server/request_handler/default_request_handler.ts

```diff
@@ -133,8 +133,14 @@ export class DefaultRequestHandler implements A2ARequestHandler {
                 await this._sendPushNotificationIfNeeded(event);
 
                 if (options?.firstResultResolver && !firstResultSent) {
-                    if (event.kind === 'message' || event.kind === 'task') {
-                        options.firstResultResolver(event as Message | Task);
+                    let firstResult: Message | Task | undefined;
+                    if (event.kind === 'message') {
+                        firstResult = event as Message;
```

Comment:
nit: `as Message` shouldn't be mandatory here, as `event.kind === 'message'` above defines the type.